addon/src/lib/Telemetry.js

```diff
@@ -79,4 +82,31 @@ export default class Telemetry {
     });
     req.post();
   }
+
+  sendPingCentreEvent(object: any, event: string, time?: number, payload: Object) {
+    // A little work is required to replicate the ping sent to Telemetry
+    // via the `submitExternalPing('testpilot', payload, opts)` call:
+    const pcPing = TelemetryController.getCurrentPingData();
+    pcPing.type = 'testpilot';
+    pcPing.payload = payload;
+    const pcPayload = {
+      event_type: event,
+      object: object,
+      client_time: makeTimestamp(time),
+      addon_id: self.id,
+      addon_version: self.version,
+      firefox_version: pcPing.environment.build.version,
+      os_name: pcPing.environment.system.os.name,
+      os_version: pcPing.environment.system.os.version,
+      locale: pcPing.environment.settings.locale,
+      raw: JSON.stringify(pcPing),
+      // Note: these two keys are normally inserted by the ping-centre client.
+      client_id: ClientID.getCachedClientID(),
+      topic: 'testpilot'
+    };
+
+    Services.appShell.hiddenDOMWindow.navigator.sendBeacon(
```

Comment:
@dannycoates sendBeacon will wait till the browser networking resources are idle before trying to send the ping